File: le-spark/src/main/resources/scripts/reinitialize.scala
rawInput = mapper.readValue[List[JsonNode]]("""{{INPUT}}""")

scriptInput = rawInput map { input => {
  val storage = input.get("StorageType").asText().toLowerCase()
  storage match {
    case "hdfs" => loadHdfsUnit(input)
    case _ => throw new UnsupportedOperationException(s"Unknown storage $storage")
  }
}
}

println("----- BEGIN SCRIPT OUTPUT -----")
println(s"Input: $scriptInput")
println("----- END SCRIPT OUTPUT -----")

scriptParams = mapper.readValue[JsonNode]("""{{PARAMS}}""")

println("----- BEGIN SCRIPT OUTPUT -----")
println(s"Params: $scriptParams")
println("----- END SCRIPT OUTPUT -----")

scriptTargets = mapper.readValue[List[ObjectNode]]("""{{TARGETS}}""")

println("----- BEGIN SCRIPT OUTPUT -----")
println(s"Targets: $scriptTargets")
println("----- END SCRIPT OUTPUT -----")

lattice = new LatticeContext(scriptInput, scriptParams, scriptTargets)

File: le-spark/src/main/resources/scripts/reinitialize.py
script_targets = json.loads('''{{TARGETS}}''')
print("----- BEGIN SCRIPT OUTPUT -----")
print("Targets:", script_targets)
print("----- END SCRIPT OUTPUT -----")

raw_input = json.loads('''{{INPUT}}''')
script_input = [load_data_unit(unit) for unit in raw_input]

print("----- BEGIN SCRIPT OUTPUT -----")
print("Input:", script_input)
print("----- END SCRIPT OUTPUT -----")

script_params = json.loads('''{{PARAMS}}''')

print("----- BEGIN SCRIPT OUTPUT -----")
print("Params: %s" % json.dumps(script_params))

lattice = LatticeContext(input=script_input, params=script_params, targets=script_targets)

